whylog/log_reader/__init__.py

```diff
@@ -21,22 +23,35 @@ def get_causes_tree(self, front_input):
 
 
 class LogReader(AbstractLogReader):
+    NO_TEMP = frozendict()
+
     def __init__(self, config):
         self.config = config
 
-    def get_causes(self, front_input):
-        input_log_type = self.config.get_log_type(front_input.line_source)
+    def get_causes(self, front_input, tmp_assign_to_log_type=NO_TEMP):
+        input_line_source = front_input.line_source
+        input_log_type = self.config.get_log_type(input_line_source)
         if not input_log_type:
-            raise NoLogTypeError(front_input)
+            input_log_type = self._get_input_log_type(tmp_assign_to_log_type, input_line_source)
         investigation_plan = self.config.create_investigation_plan(front_input, input_log_type)
         manager = SearchManager(investigation_plan)
-        return manager.investigate(front_input)
+        return manager.investigate(front_input, tmp_assign_to_log_type)
+
+    @classmethod
+    def _get_input_log_type(cls, tmp_assign_to_log_type, input_line_source):
+        for log_type, line_sources in six.iteritems(tmp_assign_to_log_type):
```

Comment:
there can be hundreds of thousands of log files in a log type. We don't want to iterate through those - we'd rather check if any of the patterns assigned to the particular log_type matches our current line_source.